platforms/fabric/src/main/java/com/dfsek/terra/fabric/data/Codecs.java

```diff
@@ -34,25 +38,36 @@
                                                                                                id)))));
     
     public static final Codec<TerraBiomeSource> TERRA_BIOME_SOURCE = RecordCodecBuilder
-            .create(instance -> instance.group(RegistryLookupCodec.of(Registry.BIOME_KEY)
-                                                                  .forGetter(TerraBiomeSource::getBiomeRegistry),
-                                               Codec.LONG.fieldOf("seed").stable()
+            .create(instance -> instance.group(RegistryCodecs.dynamicRegistry(Registry.BIOME_KEY, Lifecycle.stable(), Biome.CODEC)
+                                                             .fieldOf("biome_registry")
+                                                             .stable()
+                                                             .forGetter(TerraBiomeSource::getBiomeRegistry),
+                                               Codec.LONG.fieldOf("seed")
+                                                         .stable()
                                                          .forGetter(TerraBiomeSource::getSeed),
-                                               CONFIG_PACK.fieldOf("pack").stable()
+                                               CONFIG_PACK.fieldOf("pack")
+                                                          .stable()
                                                           .forGetter(TerraBiomeSource::getPack))
-                                        .apply(instance, instance.stable(
-                                                TerraBiomeSource::new)));
+                                        .apply(instance, instance.stable(TerraBiomeSource::new)));
     
-    public static final Codec<FabricChunkGeneratorWrapper> CODEC = RecordCodecBuilder.create(
+    public static final Codec<FabricChunkGeneratorWrapper> FABRIC_CHUNK_GENERATOR_WRAPPER = RecordCodecBuilder.create(
```

Comment:
Following the convention established with the previous fields, this should be named `TERRA_FABRIC_CHUNK_GENERATOR_WRAPPER`

Using `TERRA_CHUNK_GENERATOR_WRAPPER` is probably better, though.

Reply:
The only reason registry key codec specifies Terra is because otherwise it'd be ambiguous with Minecraft's registry key. Otherwise it's just the class name in upper snake case.